### PR TITLE
Issue 24 | ReadMe file sections and fix rendering error

### LINKED ISSUE #54: 24. update README file
## Summary
Write a ReadMe file for the project and fix some glaring errors with rendering.

### ReadMe sections:

- [x] Title + description
- [x] Why We built this: purpose & who it's for
- [x] Technologies used
- [x] Live Demo (github pages link)
- [x] Features ("User can...", etc.)
- [x] Stretch Features (if applicable)
- [x] Preview (GIF or screenshots of UI)
- [x] Testing/QA + System Requirements

### Other
- [x] Bug fixes
- [x] Center align the home page content and add padding before the title

Update 03/08: Error fixed.

~~Currently, our project in production gives this error and does not load properly.~~

tcl-error

~~Not sure if it's only me. But other teams' projects don't have this problem when I checked. Looking it up, it is related to this React error: https://reactjs.org/docs/error-decoder.html/?invariant=299~~

### PR DESCRIPTION
**Pull request by arcan9**
## Description

Our previous Read Me file that was created by TCL has been renamed `PROJECT-SETUP.md` and is only there for reference. `README.md` has been updated with details about our specific project.

## Related Issue

Closes #54 

### ReadMe sections:

- [x] Title + description
- [x] Why We built this: purpose & who it's for
- [x] Technologies used
- [x] Live Demo (github pages link)
- [x] Features ("User can...", etc.)
- [x] Stretch Features (if applicable)
- [x] Preview (GIF or screenshots of UI)
- [x] Testing/QA + System Requirements

### Other
- [x] Bug fixes
- [x] Center align the home page content and add padding before the title

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|   ✓  | :bug: Bug fix              |
|   ✓  | :scroll: Docs              |

## Testing Steps / QA Criteria

1. From your terminal, first run git pull on main and then git checkout `readme-file-and-other-error-handling`
2. **Important:** instead of npm start, do this: `npm run build` then `npm run preview`. This will open http://localhost:4173/, which will show our app _in production_. The loading error we were previously getting only occurs during production not development! So it's important to check our app in this manner.
   - **Note:** every time you change something in the code and want to check the updates in production, you always have to `npm run build` to update the build, then `npm run preview` again. The most current version in production is always based on the last time you ran the build.
 
3. You can preview what the README file will look like on Github while editing it in VSCode using the command `Ctrl+V K` when the file is open:

Screenshot 2023-03-06 143931

### CODE REVIEW

**Comment by mtfoley** [comment]
So I have next to zero exposure to the issue that y'all had been seeing, but my suspicion for the react error would related to the font-awesome icon kit being loaded in index.html. The only place I see it used is the left/right buttons in the LearnModal component. I wonder how things would work if you (perhaps in a separate PR based on main), removed the script tag for font-awesome and changed the Previous/Next buttons on the modals to just say "Prev" and "Next" instead of have an icon. If you do this and forego using the mountApp function, I wonder if the loading error would go away.

**Comment by mtfoley** [comment]
> So I have next to zero exposure to the issue that y'all had been seeing, but my suspicion for the react error would related to the font-awesome icon kit being loaded in index.html. The only place I see it used is the left/right buttons in the LearnModal component. I wonder how things would work if you (perhaps in a separate PR based on main), removed the script tag for font-awesome and changed the Previous/Next buttons on the modals to just say "Prev" and "Next" instead of have an icon. If you do this and forego using the mountApp function, I wonder if the loading error would go away.

also, it looks like we could download the SVG code for these left/right arrows individually:
e.g. https://fontawesome.com/icons/arrow-right?s=solid&f=classic

**Comment by arcan9** [comment]
> > So I have next to zero exposure to the issue that y'all had been seeing, but my suspicion for the react error would related to the font-awesome icon kit being loaded in index.html. The only place I see it used is the left/right buttons in the LearnModal component. I wonder how things would work if you (perhaps in a separate PR based on main), removed the script tag for font-awesome and changed the Previous/Next buttons on the modals to just say "Prev" and "Next" instead of have an icon. If you do this and forego using the mountApp function, I wonder if the loading error would go away.
> 
> also, it looks like we could download the SVG code for these left/right arrows individually: e.g. https://fontawesome.com/icons/arrow-right?s=solid&f=classic

@mtfoley Thanks Matt!! Looks like it’s solved by modifying the index.jsx but I’ll try your suggestion for good measure!

File: README.md [hunk 1]
```diff
@@ -1,37 +1,60 @@
...
+# Knead To Buy
 
-This README file explains The Collab Lab’s smart shopping list project and provides instructions for developing it locally.
+A full stack React app for shoppers who want to keep track of things they buy and remember how soon they need to buy them again.
 
-If you’re a Collab Lab developer, WELCOME! We’re so excited to work with you and collaborate on something amazing. Be sure to read [the setup instructions](#set-up-the-project) and then [make your first contribution](#make-your-first-contribution)! For additional details on how your cohort will operate, please check out the [project brief](PROJECT-BRIEF.md).
+## Why We Built This
 
-## What is this project?
+This project is the culmination of the collaborative efforts of four highly-skilled developers as part of [The Collab Lab](https://the-collab-lab.codes) program under the guidance of mentors. It serves as a milestone in our ability to work together and acquire new technical skills and adopt the Agile methodology.
```
Comment (Bustamor) [inline README.md:8]: Feel so fancy with you calling us "highly skilled."  Nice touch

File: src/index.jsx [hunk 2]
```diff
@@ -1,8 +1,26 @@
+/** This modified code block waits for the DOM to fully load
+ * before mounting the React app to the `root` element.
+ * Seems to solve the prior error we were having that prevented
+ * the app to load properyly on refreshes.
+ */
+
```
Comment (Bustamor) [inline src/index.jsx:6]: Thank you for solving this!  Glad you got it sorted out.  Would love to know how you figured it out.

**Review by Bustamor — COMMENTED** [review]
Beautiful work on this.  Thanks for getting this together!

File: src/index.jsx (hunk 2 above)
Comment (arcan9) [reply src/index.jsx:6]: @Bustamor React docs gave a helpful hint of the error: “Target container is not a DOM element”

Lots of googling and stack overflowing later, the common answer was that the DOM was not fully loaded and React couldn’t find it. Then, ChatGPT helped in reminding me how to listen for the DOM in vanilla Javascript! Haha 😁

Though, I’m curious to try out Matt’s solution (in a different branch separate from this one).

**Review by alex-andria — APPROVED** [review]
Looks great Ash! Thanks for taking the time to make our README pretty and solving our errors with the `root`!

**Pull request merged**